src/core/Elsa.Abstractions/Extensions/ServiceCollectionExtensions.cs

```diff
@@ -8,6 +10,45 @@ public static class ServiceCollectionExtensions
     {    
         public static IServiceCollection AddWorkflowProvider<T>(this IServiceCollection services) where T : class, IWorkflowProvider => services.AddTransient<IWorkflowProvider, T>();
         public static IServiceCollection AddTriggerProvider<T>(this IServiceCollection services) where T : class, ITriggerProvider => services.AddTransient<ITriggerProvider, T>();
+
+        /// <summary>
+        /// Add all TriggerProviders (<see cref="ITriggerProvider"/>) that are in the assembly
+        /// </summary>
+        /// <param name="services"></param>
+        /// <param name="assembly"></param>
+        /// <returns></returns>
+        public static IServiceCollection AddTriggerProvider(this IServiceCollection services, Assembly assembly)
```

Comment:
Since Elsa already depends on Scrutor, can we use that package to register services instead?

Reply:
I would like to avoid external libraries if possible. Especially if only one line is needed. I already had the problem that with external libraries both had the same dependency, but different versions. This caused conflicts and cost me days.

In addition, Scrutor is not needed. It is only used in one place AutoMapperServiceCollectionExtensions. But it makes no sense here because AutoMapper already includes a scanner https://docs.automapper.org/en/stable/Configuration.html . I would prefer the scanner from Automapper.

Reply:
Actually, Scrutor is also used by Elsa.Core for decorating services, so we might as well take advantage of it. If it causes versioning issues with consuming applications and libraries, we can always consider making it a private dependency. If it really were just a couple of lines of code, it would be OK to remove Scrutor as a dependency, but I think it;s a little bit more than that, in which case I prefer to maintain as little code as I can.

Reply:
I could take a look at why it is needed. Possibly can be replaced by a small change. If it is more, I agree with you.

Reply:
Sounds good, I'll leave this to you until you let me know otherwise. Thank you!